packages/fbjs/src/__forks__/Promise.native.js

```diff
@@ -18,8 +18,13 @@ require('promise/setimmediate/done');
 /**
  * Handle either fulfillment or rejection with the same callback.
  */
-Promise.prototype.finally = function(onSettled) {
-  return this.then(onSettled, onSettled);
-};
+Promise.prototype.finally = function(callback) {
+  return this.then(
+    value => Promise.resolve(callback()).then(() => value),
+    reason => Promise.resolve(callback()).then(() => {
+      throw reason;
+    })
+  );
+}
```

Comment:
FWIWI: We can replace this whole definition with just a call to `require('promise/setimmediate/finally')`. https://github.com/then/promise/blob/7.3.1/src/finally.js

What you have is virtually identical, just reducing the amount of code we need to have written ourselves.